src/macaron/slsa_analyzer/provenance/witness/__init__.py

```diff
@@ -77,7 +77,6 @@ def is_witness_provenance_payload(
     bool
         ``True`` if the payload is a witness provenance payload, ``False`` otherwise.
     """
-    # TODO: add support for in-toto v1 provenances.
```

Comment:
Why is this TODO removed?

Reply:
Witness provenances only use `in-toto` v0.1 at the moment. Adding support for `v1` is therefore meaningless. I don't believe the spec for Witness `v1` has been released yet, but even if it has and it uses `in-toto` v1 (or will), this TODO is not required as the changes required for supporting that will clearly require changes here regardless.

Reply:
The context of why this TODO is here in the first place was that `predicateType` for a Witness provenance is "configurable". The `predicate_types` set is from the user ini config, which was not my decision. Otherwise, I could just have hard-code the corresponding `predicateType` here and not have the TODO which was how I was doing it initially, because this function is only meant for v0.1 (and I would agree with your points here in that case), but it is not, and hence the TODO.

In addition, this file also includes similar TODO comments in a couple of places. If this comment is removed, then we can make the case to remove other TODOs as well, which I am unsure of.

I think the best thing to do is to leave the TODOs as is and concretely decide on what to do with these Witness provenances later. In general, these TODOs are less about whether v1 is currently supported or we should support v1, but more about the fact that `predicateType` is configurable.

Reply:
Given that the way in which the `in-toto` spec is used depends upon the provenance it is contained within, I believe that this (and the other TODO in the file) provides no value. We cannot add support until we know what we are adding it for. This would be better tracked in an issue.

Reply:
I agree with you here. In that case, we should remove all similar TODOs in the file.

In connection with this, unless there is any objection from anyone else, I also want to suggest hard-coding the `predicateType` check for Witness provenances instead of making it a configuration option, which was the initial reason for all of these TODOs. If we know for sure we are only supporting one specific version of Witness provenance, there is no reason to make it a configurable option. This is not a suggestion for this PR. If it is agreed, a fix can be created for it later.

Reply:
I agree with that also.